### ISSUE TITLE
[HOLD for payment 2021-12-27] [HOLD for payment 2021-12-15] Add "Workspace" to the Workspace menu on the left pane header on Desktop/Web and button to delete/add new workspaces

### ISSUE THREAD

**Issue by kevinksullivan** [Expensify/App#5185]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
# Problem

There are three issues we are tackling here based on the conversation below.
1. There is now `Workspace` header on web when a user is viewing the Workspace menu, making it inconsistent with mobile.
2. As discussed [here](https://github.com/Expensify/Expensify/issues/176926#issuecomment-918488952), there is no option to delete or create a new workspace from within the Workspace menu

![image](https://user-images.githubusercontent.com/22508290/132778848-ad8e2460-70f7-443d-80f0-623cb931f5b8.png)

# Solution
1. Add a `Workspace` header to web/desktop so that is consistent with mobile
2. Add a "3 dot button" to the Workspace menu, and allow a user to select from two options; `New Workspace` and `Delete Workspace`

![image](https://user-images.githubusercontent.com/22508290/133171456-52575bf4-c9b7-4e4a-a40d-f7da81c8b54b.png)

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @SofiedeVreese (`AutoAssignerTriage`), see https://stackoverflow.com/c/expensify/questions/4749 for more details.

**Comment by kevinksullivan** [comment]
Picking this up for ya @SofiedeVreese just to get it out there. Job posted

https://www.upwork.com/jobs/~01a0df2a984ccdedd6

**Comment by MelvinBot** [comment]
Triggered auto assignment to @chiragsalian (`Exported`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by Santhosh-Sellavel** [comment]
## Proposal
It enabled here only for small screen, we should remove the check
https://github.com/Expensify/App/blob/2578d3a662a1f405af6c8cabb8b2e36862206077/src/pages/workspace/WorkspaceSidebar.js#L100-L106

**Comment by Santhosh-Sellavel** [comment]
<img width="325" alt="Screenshot 2021-09-10 at 6 26 38 AM" src="https://user-images.githubusercontent.com/85645967/132781665-97c2e020-1669-4c34-8f52-8a1eb8e2c8bb.png">

I see more options ( vertical three dots...) in the workspace side bar, from the image above. What options do we want to show now?

**Comment by chiragsalian** [comment]
Hmm have you tested that plan @Santhosh-Sellavel. I have not but the reason i ask is because the component says  `HeaderWithCloseButton` so I feel like a close button would show next to the header which isnt what we want. 
So maybe the proposal would involve a change wherein `isSmallScreenWidth` is moved to be within `HeaderWithCloseButton` so that close button is only shown for mobile devices and not web & desktop. Does that make sense? Let me know if i'm mistaken.

**Comment by mananjadhav** [comment]
**Proposal**
Just removing the isSmallScreenWidth won't solve it as we don't want the close button. Hence I recommend adding a conditional element Header for large screens.

```
{isSmallScreenWidth 
     ? ( 
         <HeaderWithCloseButton 
             title={translate('workspace.common.workspace')} 
             onCloseButtonPress={() => Navigation.dismissModal()} 
         /> 
     ):
(
<Header title={translate('workspace.common.workspace')}  />
)
} 

```

**Comment by chiragsalian** [comment]
Nice yeah i like that plan, then we don't have to worry about the 3 dots too since its literally just the header. Plain and simple. I like it 🙂 

@kevinksullivan, feel to hire @mananjadhav in upwork.
 @mananjadhav feel free to create the PR.

**Comment by mananjadhav** [comment]
Thanks @chiragsalian

**Comment by Santhosh-Sellavel** [comment]
@chiragsalian I tested out Chirag. The `HeaderWithCloseButton` will show a close icon. I see more options in the Expected section as said here https://github.com/Expensify/App/issues/5185#issuecomment-916550999

The header alone we can use, we need some styling.

**Comment by shawnborton** [comment]
That being said, the three dots essentially replace the "x" icon. Style-wise, hopefully the top part of the left pane is basically identical to mobile, with the exception that the "X" button is removed. Here it is in full context:
![image](https://user-images.githubusercontent.com/2319350/132782788-6a45f648-4574-4179-95cb-4d4400d8a905.png)


Let me know if that make sense.

**Comment by shawnborton** [comment]
@kevinksullivan I know you had a separate issue for the three dots (I think?) but I wonder if it makes sense to address this together.

**Comment by chiragsalian** [comment]
@mananjadhav, maybe pause on the PR atm, until we confirm the design. It's a little different than the original expected result. I think it makes sense to tackle the three dots design here too.

**Comment by mananjadhav** [comment]
@shawnborton I am guessing the three dots are not there at the moment and will be added sometime later? Adding the three dots menu would be a part of `WorkspaceSidebar`, so shouldn't have an impact on the design if we just add the Header. Might require some basic flex styling changes.

**Comment by shawnborton** [comment]
Ideally it won't require any style changes, we should be able to reuse the header component in both the left pane and in the right pane. It's just a matter of only passing through the icon with three dots on the left for desktop. We will provide that .svg for the icon as well.

**Comment by Santhosh-Sellavel** [comment]
> @chiragsalian I tested out Chirag. The `HeaderWithCloseButton` will show a close icon. I see more options in the Expected section as said here [#5185 (comment)](https://github.com/Expensify/App/issues/5185#issuecomment-916550999)
> 
> The header alone we can use, we need some styling. I suggest we use the HeaderWithCloseButton component itself. And rename it to a more generic name. The close button can be configurable with a new prop like shouldShowCloseButton

**Comment by mananjadhav** [comment]
**Updated Proposal**

Based on @shawnborton's feedback on the requirement here's my updated proposal. We modify the `HeaderWithCloseButton`

1. Add three props: `shouldShowThreeDotsMenu` (for lack of better word, need suggestions ), `shouldShowCloseButton` and a function `onThreeDotIconPress`.
2. Add the following code:

```

props.shouldShowThreeDotsMenu && (
                    <Tooltip text={props.translate('common.more')}>

                        <TouchableOpacity
                            onPress={props.onThreeDotIconPress}
                            style={[styles.touchableButtonImage]}
                        >
                            <Icon src={ThreeDotIcon} />
                        </TouchableOpacity>
                    </Tooltip>

```

3. Update the close button section like:

```
props.shouldShowCloseButton && (
<Tooltip text={props.translate('common.close')}>
                    <TouchableOpacity
                        onPress={props.onCloseButtonPress}
                        style={[styles.touchableButtonImage, styles.mr0]}
                        accessibilityRole="button"
                        accessibilityLabel={props.translate('common.close')}
                    >
                        <Icon src={Close} />
                    </TouchableOpacity>
                </Tooltip>
)
```
 
4. Lastly, in `WorkspaceSidebar`, remove `isSmallScreenWidth` conditional render.
```
{
                            <HeaderWithCloseButton
                                title={translate('workspace.common.workspace')}
                                shouldShowCloseButton={isSmallScreenWidth}
                                onCloseButtonPress={isSmallScreenWidth ? () => Navigation.dismissModal(): null}

                                /** Guessing we don't have it right now but whenever we add */
                                shouldShowThreeDotMenu={true}
                                onThreeDotMenu={() => ourFunctionToHandleThis()}
                            />
                        }

```

**Comment by chiragsalian** [comment]
@Santhosh-Sellavel, firstly I'm unsure what to reply to your post because there is no question associated with it. It felt like you were sharing a thought. Then a little unsure why you linked my https://github.com/Expensify/App/issues/5185#issuecomment-916550999 to begin with.

So we've to still wait for @kevinksullivan to confirm he's on board with the changes.
Additionally, you mentioned `The HeaderWithCloseButton will show a close icon` in your post here but we don't want to show a close button hence I preferred manan's proposal at this time. The original design also didn't have 3 dots and was just the header which is why manan's original proposal fit the solution better.

Let me know if I'm misunderstanding something or if you have any follow-up questions for me. Plus feel free to hit me up on Expensify chat anytime for questions. My email - "chirag@expensify.com" 🙂

**Comment by chiragsalian** [comment]
@mananjadhav, I like the proposal thought. Question though, what are the options shown when the 3 dots are clicked? When I create a workspace on mobile or web I don't see it so I was wondering what is it even supposed to contain? Is it for specific pages that doesn't include the workspace page?

**Comment by mananjadhav** [comment]
Afaik,  it seems we are planning to add more menu options in Workspace on click of the button. Even I am not sure what atm, @shawnborton might be the best person to share. 

But I added those checks and icon so that whenever we plan to add it we just have to add the prop and the function handling.  Headed components stays the same.

**Comment by shawnborton** [comment]
Yeah, the three dots will launch a menu of options. That might be getting handled separately though, apologies for the confusion as I am not quite up-to-speed on how the issues were divvied up. Perhaps @kevinksullivan can confirm there.

**Comment by mananjadhav** [comment]
@kevinksullivan @chiragsalian Can you confirm what approach are we going to take ?

**Comment by kevinksullivan** [comment]
I am ok with using this PR to:
1. Add "Workspace"
2. Add the 3 dots

There was one other GH https://github.com/Expensify/Expensify/issues/176926 where we planned to have the 3 dots open a bottom tray that showed options to create another workspace or delete the existing workspace.

![image](https://user-images.githubusercontent.com/22508290/133145976-550da32b-8d04-4b10-bbaf-fc6e102120cf.png)


Should this GH also take care of that? Or does it make more sense to separate that out into its own issue, which is a follow-up to this one?

**Comment by parasharrajat** [comment]
I think it can be added to this. This **Add "Workspace" to the Workspace menu on the left pane header on Desktop/Web** issue sounds like a minor side-change, generally speaking.

**Comment by chiragsalian** [comment]
Yeah, i think it can be added here too. @kevinksullivan or @shawnborton, could you share how the options will look expanded for web as well, just so that we're all on the same page.

**Comment by chiragsalian** [comment]
Also should we have an option like "Rename Workspace" in the options? It's not very obvious from the mobile and web that clicking on the workspace name allows you to edit it.

**Comment by parasharrajat** [comment]
I would suggest `Edit Workspace`. That pencil icon on the avatar looks like that it is only for changing avatar and I don't this it's needed on this screen. it should be moved to the Edit workspace page.

**Comment by shawnborton** [comment]
I think we can discuss "Edit Workspace" or making the renaming part easier as a separate issue, though personally I think it's fine right now. 

In terms of what the options look like when expanded, I think this screenshot should cover it. On desktop, the menu launches near the icon. On mobile, we bottom-dock:
![image](https://user-images.githubusercontent.com/2319350/133148176-df977cff-47d6-400d-9391-2e11e0e35009.png)

**Comment by mananjadhav** [comment]
> In terms of what the options look like when expanded, I think this screenshot should cover it. On desktop, the menu launches near the icon. On mobile, we bottom-dock:

Yeah, this works. 

@chiragsalian I can create a new Component similar to `VideoChatButtonAndMenu`, except for the fact that our button lies outside the component. We'll also have to add ConfirmModal for the delete workspace button. Also, can you help me with the delete workspace API?

**Comment by chiragsalian** [comment]
Yup i can help you out with any of the API's, can you post your complete plan of action for me to review?

**Comment by kevinksullivan** [comment]
I went ahead and updated the OP as well, so that we're clear that this GH should:

1. Add a Workspace header to web/desktop so that is consistent with mobile
2. Add a "3 dot button" to the Workspace menu, and allow a user to select from two options; New Workspace and Delete Workspace

**Comment by chiragsalian** [comment]
@mananjadhav, any update here? Can you post your complete plan of action for me to review?

**Comment by mananjadhav** [comment]
**Updated Proposal:**

Apologies for the delay. Took a little while for the long proposal. Here's my updated proposal including the updated implementation:

1. We still modify the `HeaderWithCloseButton` with additional props: `shouldShowCloseButton` and add a `moreOptionsComponent` (renamed ThreeDots from previous proposal to MoreOptions)
2. We'll create 2 components: `WorkspaceHeader` and `WorkspaceOptionsMenu`
3. We'll replace `HeaderWithCloseButton` with `WorkspaceHeader` in `WorkspaceSidebar`

		<WorkspaceHeader policy={policy} />

4. Our `WorkspaceHeader` will look something like this:

```

class WorkspaceHeader extends React.Component {
	
	propTypes = {
		/** Policy for the current route */
	    policy: PropTypes.shape({
	        /** ID of the policy */
	        id: PropTypes.string,

	        /** Name of the policy */
	        name: PropTypes.string,
	    }),

	}

	render() {
		<View>
			<HeaderWithCloseButton
			    title={translate('workspace.common.workspace')}
			    onCloseButtonPress={() => Navigation.dismissModal()}
			    shouldShowCloseButton={isSmallScreenWidth}
			    shouldShowMoreOptionsMenu
			    moreOptionsComponent={<WorkspaceOptionsMenu policyId={policy.id} />}
			/>
		</View>

	}
}

```



5. Our main component `WorkspaceOptionsMenu` that has the popover will look something similar to `VideoChatMenu`. I've added maximum code with approach.

```

class WorkspaceOptionsMenu extends Component {
	
	props = {
		policyId: PropTypes.string
	}

    constructor(props) {
        super(props);

        this.toggleOptionsMenu = this.toggleOptionsMenu.bind(this);
        this.measurePopOverPosition = this.measurePopOverPosition.bind(this);
        this.moreOptionsWrapper = null;
        this.state = {
            popoverPosition: {x: 0, y: 0},
            isOptionsMenuOpen: false,
            isDeleteModalOpen: false
        };
    }

    componentDidMount() {
        Dimensions.addEventListener('change', this.measurePopOverPosition);
    }

    componentWillUnmount() {
        Dimensions.removeEventListener('change', this.measurePopOverPosition);
    }

    toggleOptionsMenu() {
        this.setState(prevState => ({
            isOptionsMenuOpen: !prevState.isOptionsMenuOpen,
        }));
    }

    confirmDeleteAndHideModal() {
        deleteWorkspace(this.props.policyId);
        this.hideDeleteModal();
    }

    hideDeleteModal() {
        this.setState({
            isDeleteModalOpen: false,
        });
    }


    /**
     * This gets called onLayout to find the cooridnates of the wrapper for the threedots button.
     */
    measurePopOverPosition() {
        if (this.moreOptionsWrapper) {
            this.moreOptionsWrapper.measureInWindow((x, y) => this.setState({
                popoverPosition: {x, y},
            }));
        }
    }

    render() {

    	this.menuItemData = [
            {
                icon: Plus, // imported from Icon/Expensicons
                text: props.translate('workspace.addWorkspace'),
                onPress: () => create() // Imported from src/libs/actions/Policy
            },
            {
                icon: Trashcan, // imported from Icon/Expensicons
                text: props.translate('workspace.deleteWorkspace'),
                onPress: () => this.setState({ isDeleteModalOpen: true })
            },
        ].map(item => ({
            ...item,
            onPress: () => {
                item.onPress();
                this.toggleOptionsMenu();
            },
        }));


        return (
            <>
                <View
                    ref={el => this.moreOptionsWrapper = el}
                    onLayout={this.measurePopOverPosition}
                >
                    <Tooltip text={this.props.translate('common.moreOptions')}>
                        <Pressable
                            onPress={() => {this.toggleOptionsMenu();}}
                            style={[styles.touchableButtonImage, styles.mr0]}
                        >
                            <Icon src={ThreeDots} />
                        </Pressable>
                    </Tooltip>
                </View>
                <Popover
                    onClose={this.toggleOptionsMenu}
                    isVisible={this.state.isOptionsMenuOpen}
                    anchorPosition={{
                        left: this.state.popoverPosition.x - 150,
                        top: this.state.popoverPosition.y + 40,
                    }}
                    animationIn="fadeInDown"
                    animationOut="fadeOutUp"
                >
                    {this.menuItemData.map(({icon, text, onPress}) => (
                        <MenuItem
                            wrapperStyle={styles.mr3}
                            key={text}
                            icon={icon}
                            title={text}
                            onPress={onPress}
                        />
                    ))}
                </Popover>
                <ConfirmModal
                    title={this.props.translate('workspace.deleteWorkspace')}
                    isVisible={this.state.isDeleteModalOpen}
                    onConfirm={this.confirmDeleteAndHideModal}
                    onCancel={this.hideDeleteModal}
                    prompt={this.props.translate('workspace.deleteConfirmation')}
                    confirmText={this.props.translate('common.delete')}
                    cancelText={this.props.translate('common.cancel')}
                /
            </>
        );
    }
}

WorkspaceOptionsMenu.propTypes = propTypes;
WorkspaceOptionsMenu.defaultProps = defaultProps;
WorkspaceOptionsMenu.displayName = 'WorkspaceOptionsMenu';
export default compose(
    withWindowDimensions,
    withLocalize,
)(WorkspaceOptionsMenu);


```

6. Add the `deletePolicy` function in `src/libs/actions/Policy.js`

7. Handle translations in en.js and es.js


A couple of things to highlight:

1. Because I need the popover to open next to the MoreOptions (ThreeDots), I have to pass the whole component as the position will be determined at the runtime. Other approach to this would be to have the IconButton in the `Header` and pass the reference to the `WorkapceOptionsMenu`. I generally prefer keeping the anchor and the popover in the same comoponent. But if you want to split we can do that too.

2. In addition to what I mentioned in point 1, one other suggestion would be skip `HeaderWithCloseButton` altogether and add the title, the close button and more options in `WorkspaceHeader`. This is what I'll personally recommend and I've added the updated `WorkspaceHeader` code below.

2. There's a permission that is used in FAB for workspace creation. `...(Permissions.canUseFreePlan(this.props.betas)`. Do we need to add it here too?

3. Can you let me know if there's any Permissions or checks to hide/show Delete Workspace?

4. By default the Workspace is created with name as email prefix. Here also we'll do the same right?

5. Components might require some styling changes which I'll add when I work on the implementation.

6. Lastly, I feel the budget of 250$ is less for the updated score. Can we please revise the budget then it'll be great.



**WorkspaceHeader as per point 2**

With this `WorkspaceHeader` component, we don't have to modify `HeaderWithCloseButton`.


```

class WorkspaceHeader extends React.Component {
	
	propTypes = {
		/** Policy for the current route */
	    policy: PropTypes.shape({
	        /** ID of the policy */
	        id: PropTypes.string,

	        /** Name of the policy */
	        name: PropTypes.string,
	    }),

	}

	render() {
   		<View style={[styles.headerBar, props.shouldShowBorderBottom && styles.borderBottom]}>

   			<View style={[
	            styles.dFlex,
	            styles.flexRow,
	            styles.alignItemsCenter,
	            styles.flexGrow1,
	            styles.justifyContentBetween,
	            styles.overflowHidden,
	        ]}
	        >


			<Header title={translate('workspace.common.workspace')} />
			<WorkspaceOptionsMenu policyId={policy.id} />
			{
				isSmallScreenWidth && (
					<Tooltip text={props.translate('common.close')}>
	                    <TouchableOpacity
	                        onPress={() => Navigation.dismissModal()}
	                        style={[styles.touchableButtonImage, styles.mr0]}
	                        accessibilityRole="button"
	                        accessibilityLabel={props.translate('common.close')}
	                    >
	                        <Icon src={Close} />
	                    </TouchableOpacity>
	                </Tooltip>
				)

			}
			
			/>
		</View>

	}
}

```

cc - @chiragsalian

**Comment by chiragsalian** [comment]
Cool, the updated plan looks good to me. Some thoughts though, it feels like the `WorkspaceOptionsMenu` is a nice opportunity to set it up in a reusable manner. How about we create it as `PopupMenu` which accepts any icon. Then for this solution when you call `HeaderWithCloseButton`  pass a prop like `shouldShowThreeDotsButton` (similar to what you have but I think I like ThreeDotsButton since its more verbose) as true also pass `menuItemData` with the necessary callbacks. You've to update `HeaderWithCloseButton` to call `PopupMenu` with `ThreeDotsIcon` image passed as the default from `HeaderWithCloseButton`. Finally, with those changes, I don't think `WorkspaceHeader ` is really needed and instead, you can call `HeaderWithCloseButton` directly within `WorkspaceSidebar`. I feel like this would clean up the code and have `PopupMenu` as a reusable generic component. You could even DRY the code in VideoChatButtonAndMenu by reusing this component with the only change being the icon.
Thoughts?

Answering your "A couple of things to highlight",
1. It's fine to me the way you currently mentioned it in your plan.
2. I personally would rather re-use `HeaderWithCloseButton` with my proposed changes above it would be even more closely dependent on `HeaderWithCloseButton`.
3. No not for now. Correct me if I'm wrong @kevinksullivan, only if you're under free plan beta can you create a workspace correct? Can you ever see a workspace if you are not in the free plan beta?
4. Nope no permissions to check. Atm, I believe only admins of the workspace see these workspaces as an option to click into. Correct me if I'm mistaken @kevinksullivan.
5. Yup.
6. Sure
7. I'll leave that part to @kevinksullivan too 🙂 

Also, I'm unsure if you read but if not this issue is under n6-hold. That means even if you create the PR we cannot merge it until the label is removed. Just letting you know, you can still work on the issue and I can review it though.

**Comment by mananjadhav** [comment]
@chiragsalian I am fine with the approach of having `HeaderWithCloseButton` and making a generic `PopupMenu` too if that will have more use-cases to handle :)

**Comment by chiragsalian** [comment]
Cool we can wait for @kevinksullivan to answer some of your questions but he might be a little busy at the moment working on n6 stuff so I'm unsure when he can get to this. Either way, the proposal sounds good to me 👍  
From my perspective, @kevinksullivan can hire you (@mananjadhav) on Upwork and you can create the PR.

**Comment by mananjadhav** [comment]
Will wait for @kevinksullivan's response for this one.

**Comment by mananjadhav** [comment]
@kevinksullivan @chiragsalian Do we have any updates for me here?

**Comment by mananjadhav** [comment]
Thanks @kevinksullivan for the answers. Proposal submitted

**Comment by kevinksullivan** [comment]
Hired @mananjadhav

**Comment by chiragsalian** [comment]
Code is being worked on. There is a draft PR with good amount of progress in it.

**Comment by kevinksullivan** [comment]
@mananjadhav what's the latest on this one?

**Comment by botify** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.1.18-3 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/5921


If no regressions arise, payment will be issued on 2021-12-15. :confetti_ball:

**Comment by mananjadhav** [comment]
@chiragsalian https://expensify.slack.com/archives/C01GTK53T8Q/p1639142507005200 Should we treat this as a regression from my PR. 

The problem is `this.menuItems` are initialized in the constructor and hence they don't get the updated values. We'll have to define them in the `render` method so that they receive the latest props.

This wasn't the case when we had the old layout and I had explicitly tested this.

**Comment by chiragsalian** [comment]
> Should we treat this as a regression from my PR.

Sure, yeah I feel like I tested this too in your PR at some point. But I guess i lot changed around the N6 stuff. Anyway thanks for creating the follow-up PR.

**Comment by mananjadhav** [comment]
@kevinksullivan  Bump on Upwork payment.

**Comment by mallenexpensify** [comment]
Paid @mananjadhav in Upwork, wanted to get it issued before the weekend. Job was $250 and I added $250 because the PR was during the Company Offsite Hold (where I forgot to properly label all issues that were eligble).   If anything doesn't look correct, please comment and/or reopen.

**Comment by mananjadhav** [comment]
@mallenexpensify This was a bigger task and we had decided to add bonus completion based on the scope. For your reference, https://github.com/Expensify/App/issues/5185#issuecomment-928735294 

and this issue is also eligible for n6-hold. Primarily all the rework happened due to n6-hold only.

**Comment by botify** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.1.21-1 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/6697


If no regressions arise, payment will be issued on 2021-12-27. :confetti_ball:

**Comment by mallenexpensify** [comment]
Sorry @mananjadhav , I think I checked for the n6 hold but missed because it was listed in 'hidden items' in the issue (I really wish those could be auto-extracted every time) 

So.. $250 for the [#5185 (comment)](https://github.com/Expensify/App/issues/5185#issuecomment-928735294) and $250 for n6-hold, correct?

**Comment by mananjadhav** [comment]
That’s right @mallenexpensify

**Comment by mallenexpensify** [comment]
Paid @mananjadhav $500! Thanks!
